src/code_review_agent/integrations/github_client.py

```diff
@@ -0,0 +1,397 @@
+                    
+                    if content.encoding == "base64":
+                        decoded_content = base64.b64decode(content.content).decode('utf-8')
+                    else:
+                        decoded_content = content.content
+                    
+                    file_contents[file_change.filename] = decoded_content
+                    
+                except GithubException as e:
+                    if e.status == 404:
+                        logger.warning(
+                            "File not found",
+                            repo=repo_full_name,
+                            filename=file_change.filename,
+                            ref=ref
+                        )
+                    else:
+                        logger.error(
+                            "Error retrieving file content",
+                            repo=repo_full_name,
+                            filename=file_change.filename,
+                            ref=ref,
+                            error=str(e)
+                        )
+                except UnicodeDecodeError:
+                    logger.warning(
+                        "Binary file skipped",
+                        repo=repo_full_name,
+                        filename=file_change.filename
+                    )
+            
+            logger.debug(
+                "Retrieved file contents",
+                repo=repo_full_name,
+                files_retrieved=len(file_contents),
+                total_files=len(file_changes)
+            )
+            
+            return file_contents
+            
+        except GithubException as e:
+            logger.error(
+                "Error retrieving file contents",
+                repo=repo_full_name,
+                ref=ref,
+                error=str(e)
+            )
+            raise
+    
+    async def post_review_comments(
+        self,
+        repo_full_name: str,
+        pr_number: int,
+        comments: List[ReviewComment],
+        summary: str,
+        commit_sha: str
+    ) -> bool:
+        """Post review comments to a pull request."""
+        try:
+            pr = await self.get_pull_request(repo_full_name, pr_number)
+            
+            # Group comments by file for better organization
+            comments_by_file = {}
+            for comment in comments:
+                if comment.file_path not in comments_by_file:
+                    comments_by_file[comment.file_path] = []
+                comments_by_file[comment.file_path].append(comment)
+            
+            # Post individual line comments
+            review_comments = []
+            for file_path, file_comments in comments_by_file.items():
+                for comment in file_comments:
+                    if comment.line_number:
+                        review_comment = {
+                            "path": file_path,
+                            "line": comment.line_number,
```

Comment:
GitHub API requires a `side` when using `line`-anchored review comments on a commit diff. Without it, the review creation can fail with a 422. Consider adding `side="RIGHT"` for new/modified lines (and handling multi-line ranges when `start_line` is present).

```suggestion
review_comment = {
    "path": file_path,
    "line": comment.line_number,
    "side": "RIGHT",
    "body": self._format_comment_body(comment)
}
```